src/components/Meme.js

```diff
@@ -81,6 +88,56 @@ export default function Meme() {
 		}
 	}
 
+	
+	//redering the text data in h2 tag 
+	const memeTexts = memeTextData.map(text => {
+		return <h2
+				key={text.id}
+				draggable
+				onDragStart={(e)=>handleDragStart(e, text.id)} 
+				style={text.style}
+				className="meme__text"
+			> 
+				{text.id === "1" ? meme.topText : meme.bottomText} 
+			</h2>
+	})
+	
+	//when the text starts to drag 
+	const handleDragStart = (e, id) => {
+		e.dataTransfer.setData("id",id);
+	}
+
+	//when the text is dragged over the meme image
+	const handleDragOver = (e) => {
+		e.preventDefault();
+	}
+
+	//when the text is dropped on the meme image
+	const handleDrop = (e) => {
+		e.preventDefault();
+		//get the id that we set during dragstart
+		const id = e.dataTransfer.getData("id");
+		//get the xy coordinates of the mouse
+		let x = e.clientX;
+		let y = e.clientY;
+		//get the rectangle coordinates of the meme image.
+		const rect = document.querySelector('.meme').getBoundingClientRect();
+
+		//create a temp text array to make changes in the style and then set it to original state
+		const tempTexts = [...memeTextData]
+		//change the style of selected text and position it on the cursor
+		let index = tempTexts.findIndex((text) => text.id === id);
+		if(index == -1) return;
+		tempTexts[index].style = {
+			//distance of text from left of image is equal to dist of the cursor from margin - meme image from left.
+			left: `${x-rect.left}px`,
+			//similarly calculate top.
+			top: `${y-rect.top}px`,
```

Comment:
it gives little inconsistent effect, check online to improve it, may try to consider the width of the text

Reply:
i cannot find a better way to implement the effect. can you help me out a little in it

Reply:
try this ,maybe this could help [link](https://www.w3schools.com/howto/howto_js_draggable.asp)